src/commands/lambda/uninstrument.ts

```diff
@@ -2,27 +2,74 @@ import {CloudWatchLogs, Lambda} from 'aws-sdk'
 import {bold, cyan, red, yellow} from 'chalk'
 import {Command} from 'clipanion'
 import {parseConfigFile} from '../../helpers/utils'
-import {collectFunctionsByRegion, updateLambdaFunctionConfigs} from './functions/commons'
+import {AWS_DEFAULT_REGION_ENV_VAR} from './constants'
+import {
+  collectFunctionsByRegion,
+  getAllLambdaFunctionConfigs,
+  isMissingAWSCredentials,
+  updateLambdaFunctionConfigs,
+  willUpdateFunctionConfigs,
+} from './functions/commons'
 import {getUninstrumentedFunctionConfigs, getUninstrumentedFunctionConfigsFromRegEx} from './functions/uninstrument'
 import {FunctionConfiguration} from './interfaces'
+import {requestAWSCredentials, requestChangesConfirmation, requestFunctionSelection} from './prompt'
 
 export class UninstrumentCommand extends Command {
   private config: any = {
     functions: [],
-    region: process.env.AWS_DEFAULT_REGION,
+    region: process.env[AWS_DEFAULT_REGION_ENV_VAR],
   }
   private configPath?: string
   private dryRun = false
   private forwarder?: string
   private functions: string[] = []
+  private interactive = false
   private regExPattern?: string
   private region?: string
 
   public async execute() {
     const lambdaConfig = {lambda: this.config}
     this.config = (await parseConfigFile(lambdaConfig, this.configPath)).lambda
 
-    const hasSpecifiedFuntions = this.functions.length !== 0 || this.config.functions.length !== 0
+    let hasSpecifiedFuntions = this.functions.length !== 0 || this.config.functions.length !== 0
+    // Trial user experience
+    if (this.interactive) {
+      try {
+        if (isMissingAWSCredentials()) {
+          this.context.stdout.write(`${bold(yellow('[!]'))} AWS Credentials are missing, let's set them up!\n`)
+          await requestAWSCredentials()
+        }
+      } catch (e) {
+        this.context.stdout.write(`${red('[Error]')} ${e}\n`)
+
+        return 1
+      }
+
+      const region = this.region ?? this.config.region ?? process.env[AWS_DEFAULT_REGION_ENV_VAR]
+      this.region = region
+
+      if (!hasSpecifiedFuntions) {
+        try {
+          const lambda = new Lambda({region})
+          this.context.stdout.write('Fetching lambda functions, this might take a while.\n')
```

Comment:
Should we show the customer what region we are searching here?

Reply:
(unless they ALWAYS need to specify a region beforehand. I am thinking about a scenario where we use a default region and don't tell them)

Reply:
We probably could, but this only happens when the user has the environment variable `AWS_DEFAULT_REGION` set. Which normally doesn't happen since most of the users would directly pipe the role with the desired region. If no region is set and that environment variable is not exported, we always ask them to set it. 
Nonetheless, when instrumenting interactively, the user will be required to prompt the desired region (if not previously set by, lets say AWS_VAULT).

Reply:
OK, sounds good!